storage/innobase/include/lock0lock.h

```diff
@@ -697,6 +698,31 @@ class lock_sys_t
 
   /** mutex covering lock waits; @see trx_lock_t::wait_lock */
   MY_ALIGNED(CPU_LEVEL1_DCACHE_LINESIZE) mysql_mutex_t wait_mutex;
+
+  template<bool spinloop = false>
+  void wait_mutex_lock()
+  {
+    if (spinloop)
+    {
+      if (!mysql_mutex_trylock(&wait_mutex))
+        return;
+
+      for (auto spin= srv_n_spin_wait_rounds; spin; spin--)
+      {
+        ut_delay(srv_spin_wait_delay);
+        if (!mysql_mutex_trylock(&wait_mutex))
+         return;
+      }
+    }
```

Comment:
Can you replace the `for` loop with a call to an `ATTRIBUTE_NOINLINE` function, say, `spin_for_mutex(&wait_mutex);`?

The reasons why I suggest that are:
1. Less object code duplication (bloat), hopefully better performance
1. Easier to catch the spinning in `perf` reports.

That function could be located in the file `srw_lock.cc` and make use of the functions `srw_pause_delay()` and `srw_pause()`.